scipy/optimize/zeros.c

```diff
@@ -110,7 +110,7 @@ call_solver(solver_type solver, PyObject *self, PyObject *args)
     if (!setjmp(env)) {
         /* direct return */
         memcpy(params.env, env, sizeof(jmp_buf));
-        params.error_num = 0;
+        // error_nums must be initialized to zero by the solver
```

Comment:
Note that this is the first usage of a post-C89 style comment in this file.

Reply:
I might have misunderstood. Is this is good thing or a bad thing? Should I change it to this?

    /* error_nums must be initialized to zero by the solver */

Reply:
It may not matter too much for SciPy if the CI is happy I guess.

Reply:
Yep, best to change this to `/* ... */`. We follow the NumPy C style guide, and have never allowed `//` style comments. See https://github.com/numpy/numpy/blob/master/doc/C_STYLE_GUIDE.rst.txt#c-dialect

Reply:
Unclear about the history of the portability statement on `//`, but seeing for example the recent AIX and Solaris build issues, it wouldn't surprise me that it's a real issue. CI won't catch that.

Reply:
✔️ done

Reply:
I suspect it can happen eventually -- I think the 2.7 compiler situation on Windows is kind of bad with a C89 requirement typical if I'm not mistaken.